modules/loader/src/main/java/com/opengamma/strata/loader/csv/RatesCalibrationCsvLoader.java

```diff
@@ -0,0 +1,368 @@
+                Messages.format("Unknown RateCurveId subclass: {}", curveUse.getClass().getName()));
+          }
+        }
+      }
+    }
+    return resultBuilder.entrySet().stream()
+        .collect(toImmutableMap(e -> e.getKey(), e -> e.getValue().build()));
+  }
+
+  //-------------------------------------------------------------------------
+  // loads a single curves CSV file
+  // requestedDate can be null, meaning load all dates
+  private static Map<CurveName, NodalCurveDefinition> loadSingle(
+      ResourceLocator resource,
+      Map<CurveName, LoadedCurveSettings> settingsMap) {
+
+    CsvFile csv = CsvFile.of(resource.getCharSource(), true);
+    Map<CurveName, List<CurveNode>> allNodes = new HashMap<>();
+    for (int i = 0; i < csv.rowCount(); i++) {
+      String curveNameStr = csv.field(i, CURVE_NAME);
+      String label = csv.field(i, CURVE_LABEL);
+      String symbologyStr = csv.field(i, CURVE_SYMBOLOGY);
+      String tickerStr = csv.field(i, CURVE_TICKER);
+      String fieldNameStr = csv.field(i, CURVE_FIELD_NAME);
+      String typeStr = csv.field(i, CURVE_TYPE);
+      String conventionStr = csv.field(i, CURVE_CONVENTION);
+      String timeStr = csv.field(i, CURVE_TIME);
+      String spreadStr = csv.field(i, CURVE_SPREAD);
+
+      CurveName curveName = CurveName.of(curveNameStr);
+      StandardId id = StandardId.of(symbologyStr, tickerStr);
+      FieldName fieldName = fieldNameStr.isEmpty() ? FieldName.MARKET_VALUE : FieldName.of(fieldNameStr);
+      QuoteKey quoteKey = QuoteKey.of(id, fieldName);
+      double spread = spreadStr.isEmpty() ? 0d : Double.parseDouble(spreadStr);
+
+      List<CurveNode> curveNodes = allNodes.computeIfAbsent(curveName, k -> new ArrayList<CurveNode>());
+      curveNodes.add(createCurveNode(typeStr, conventionStr, timeStr, label, quoteKey, spread));
+    }
+    return buildCurveDefinition(settingsMap, allNodes);
+  }
+
+  // build the curves
+  private static Map<CurveName, NodalCurveDefinition> buildCurveDefinition(
+      Map<CurveName, LoadedCurveSettings> settingsMap,
+      Map<CurveName, List<CurveNode>> allNodes) {
+
+    ImmutableMap.Builder<CurveName, NodalCurveDefinition> results = ImmutableMap.builder();
+    for (Map.Entry<CurveName, List<CurveNode>> entry : allNodes.entrySet()) {
+      CurveName name = entry.getKey();
+      LoadedCurveSettings settings = settingsMap.get(name);
+      if (settings == null) {
+        throw new IllegalArgumentException(Messages.format("Missing settings for curve: {}", name));
+      }
+      results.put(name, settings.createCurveDefinition(entry.getValue()));
+    }
+    return results.build();
+  }
+
+  // create the curve node
+  private static CurveNode createCurveNode(
+      String typeStr,
+      String conventionStr,
+      String timeStr,
+      String label,
+      QuoteKey quoteKey,
+      double spread) {
+
+    if ("FIX".equals(typeStr) || "IborFixingDeposit".equals(typeStr)) {
```

Comment:
I expect this code to be rewritten relatively soon (hence my comment in the PR description), so it doesn't seem worth doing this now.